crates/hdx_parser/src/comparison.rs

```diff
@@ -13,25 +14,28 @@ pub enum Comparison {
 
 impl<'a> Parse<'a> for Comparison {
 	fn parse(parser: &mut Parser<'a>) -> Result<Comparison> {
-		Ok(match parser.next() {
-			Token::Delim('=') => Comparison::Equal,
-			Token::Delim('>') => {
-				if let Token::Delim('=') = parser.peek_with(Include::Whitespace) {
-					parser.advance_with(Include::Whitespace);
-					Comparison::GreaterThanEqual
-				} else {
-					Comparison::GreaterThan
+		let next = parser.next();
+		Ok(match_delim! {parser.next() :
+		  '=' => Comparison::Equal,
+		  '>' => {
+				match_delim!{ parser.peek_with(Include::Whitespace) :
+					'=' => {
+					   parser.advance_with(Include::Whitespace);
+					   Comparison::GreaterThanEqual
+					},
+					_ =>  Comparison::GreaterThan
 				}
-			}
-			Token::Delim('<') => {
-				if let Token::Delim('=') = parser.peek_with(Include::Whitespace) {
-					parser.advance_with(Include::Whitespace);
-					Comparison::LessThanEqual
-				} else {
-					Comparison::LessThan
+			},
+			'<' => {
+				match_delim!{ parser.peek_with(Include::Whitespace) :
+					'=' => {
+								parser.advance_with(Include::Whitespace);
+								Comparison::LessThanEqual
+					},
+					_ => Comparison::LessThan
 				}
-			}
-			token => unexpected!(parser, token),
+			},
+			_ =>  unexpected!(parser, next)
```

Comment:
Given above you'll need to:

```suggestion
			token =>  unexpected!(parser, token)
```